ext/BlockSparseArraysTensorAlgebraExt/BlockSparseArraysTensorAlgebraExt.jl

```diff
@@ -94,13 +94,17 @@ function TensorAlgebra.splitdims(
     groupreducewhile(tensor_product, split_axes, ndims(a); init=OneToOne()) do i, axis
       return length(axis) ≤ length(axes(a, i))
     end
-  blockperms = invblockperm.(blocksortperm.(axes_prod))
+  blockperms = blocksortperm.(axes_prod)
+  sorted_axes = ntuple(
+    i -> gradedrange(map(b -> length(axes_prod[i][b]), blockperms[i])), ndims(a)
+  )
```

Comment:
```suggestion
map((r, I) -> r[I], axes_prod, blockperms)
```
Does that do the same thing? You should be able to index a `GradedUnitRange` by a `Vector{<:Block{1}}` to do a block permutation.

Reply:
Ah, I got a bit confused, I think it would be:
```suggestion
  sorted_axes = map((r, I) -> only(axes(r[I])), axes_prod, blockperms)
```
i.e.:
```julia
julia> using BlockArrays, GradedUnitRanges, SymmetrySectors

julia> r = gradedrange([U1(0) => 2, U1(1) => 3])
GradedUnitRanges.GradedOneTo{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}, Vector{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}}}
U(1)[0] => 1:2
U(1)[1] => 3:5

julia> r[[Block(2), Block(1)]]
2-blocked 5-element BlockVector{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}, Vector{LabelledNumbers.LabelledUnitRange{Int64, UnitRange{Int64}, U1{Int64}}}, Tuple{GradedUnitRanges.GradedOneTo{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}, Vector{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}}}}}:
 LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(3, U(1)[1])
 LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(4, U(1)[1])
 LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(5, U(1)[1])
 ─────────────────────────────────────────────────────────────
 LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(1, U(1)[0])
 LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(2, U(1)[0])

julia> only(axes(r[[Block(2), Block(1)]]))
GradedUnitRanges.GradedOneTo{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}, Vector{LabelledNumbers.LabelledInteger{Int64, U1{Int64}}}}
U(1)[1] => 1:3
U(1)[0] => 4:5
```
Not sure if that is better than what you wrote, though what you wrote could be simplified a bit to:
```suggestion
  sorted_axes = map((r, I) -> gradedrange(length.(blocks(r[I]))), axes_prod, blockperms)
```
(ideally we could use `blocklengths` for that but that isn't defined for `BlockVector` right now).

Reply:
Since we plan to no longer attach the sector information to the block lengths, probably this version is better:
```suggestion
  sorted_axes = map((r, I) -> only(axes(r[I])), axes_prod, blockperms)
```
since it is more "future proof".